app/app.gradle.kts

```diff
@@ -39,6 +40,23 @@ tasks.withType<Jar> {
 }
 
 /**
+ * Copies the built jar(s) from the test_plugins project into the test resources directory.
+ */
+val copyTestPlugins = task<Copy>("copyTestPlugins") {
+    description = "Copies the built jar(s) from the test_plugins project into the test resources directory"
+    dependsOn(project("test_plugins").tasks["jar"])
+    from("""${project("test_plugins").buildDir}/libs""")
+    into("src/test/resources")
```

Comment:
Copy this into a `build` directory and add an additional test resource using the gradle API.
Then it will get removed when you run a `./gradlew clean`.

Reply:
Adding `test_plugins/build/libs` as a test resource directory avoids copying and also lets `gradlew clean` remove the JAR.